data/mods/gen4/moves.ts

```diff
@@ -1302,6 +1302,23 @@ export const Moves: import('../../../sim/dex-moves').ModdedMoveDataTable = {
 			},
 		},
 	},
+	psychoshift: {
+		inherit: true,
+		onTryHit(target, source, move) {
+			if (!source.status) return false;
+			source.addVolatile('psychoshift');
+			move.status = source.status;
+		},
+		self: null,
+		condition: {
+			duration: 1,
+			onAnySetStatus(status, target, source, effect) {
```

Comment:
``onSourceSetStatus`` is best here.

Reply:
Also, ``effect`` can be confusing in the context of the ``SetStatus`` event. ``sourceEffect`` is a more explicit name.